packages/taco/src/conditions/context/context.ts

```diff
@@ -86,16 +88,24 @@ export class ConditionContext {
   }
 
   private validateAuthProviders(requestedParameters: Set<string>): void {
-    requestedParameters
-      .forEach(param => {
-        const maybeAuthMethod = AUTH_METHOD_FOR_PARAM[param];
-        if (!maybeAuthMethod) {
-          return;
-        }
-        if (!this.authProviders[maybeAuthMethod]) {
-          throw new Error(ERR_AUTH_PROVIDER_REQUIRED(param));
-        }
-      });
+    for (const param of requestedParameters) {
+      // If it's not a user address parameter, we can skip
+      if (!USER_ADDRESS_PARAMS.includes(param)) {
+        continue;
+      }
+
+      // If it's a user address parameter, we need to check if we have an auth provider
+      const authMethod = AUTH_METHOD_FOR_PARAM[param];
+      if (!authMethod && !this.customParameters[param]) {
```

Comment:
For this case, should we also check that the `customParameters[param]` value is an `AuthSignature`? So not just that it is present in the custom parameters, but its type is correct. Or is that already done somewhere else?

Reply:
I don't think we validate it against the structure of `AuthSignature`. We may want to use `zod` for that. Let's see.

Reply:
Documented in https://github.com/nucypher/taco-web/issues/535, will address in a follow-up PR